csharp/src/Ice/Communicator.cs

```diff
@@ -1404,6 +1446,216 @@ private static void LoadReferencedAssemblies(Assembly a)
+                    {
+                        // Ignored
+                        plugins.Remove(key);
+                    }
+                    else if (suffix == "clr")
+                    {
+                        name = name.Substring(0, dotPos);
+                        LoadPlugin(name, val, ref cmdArgs);
+                        plugins.Remove(key);
+                        plugins.Remove($"Ice.Plugin.{name}");
+                    }
+                    else
+                    {
+                        // Name is just a regular name that happens to contain a dot
+                        dotPos = -1;
+                    }
+                }
+
+                if (dotPos == -1)
+                {
+                    plugins.Remove(key);
+
+                    // Is there a .clr entry?
+                    string clrKey = $"Ice.Plugin.{name}.clr";
+                    if (plugins.ContainsKey(clrKey))
+                    {
+                        val = plugins[clrKey];
+                        plugins.Remove(clrKey);
+                    }
+                    LoadPlugin(name, val, ref cmdArgs);
+                }
+            }
+
+            void LoadPlugin(string name, string pluginSpec, ref string[] cmdArgs)
+            {
+                string[] args = Array.Empty<string>();
+                string? entryPoint = null;
+                if (pluginSpec.Length > 0)
+                {
+                    // Split the entire property value into arguments. An entry point containing spaces must be enclosed
+                    // in quotes.
+                    try
+                    {
+                        args = Options.Split(pluginSpec);
+                    }
+                    catch (FormatException ex)
+                    {
+                        throw new InvalidConfigurationException($"invalid arguments for plug-in `{name}'", ex);
+                    }
+
+                    Debug.Assert(args.Length > 0);
+
+                    entryPoint = args[0];
+
+                    args = args.Skip(1).ToArray();
+
+                    // Convert command-line options into properties. First we convert the options from the plug-in
+                    // configuration, then we convert the options from the application command-line.
+                    var properties = new Dictionary<string, string>();
+                    properties.ParseArgs(ref args, name);
+                    properties.ParseArgs(ref cmdArgs, name);
+                    foreach (KeyValuePair<string, string> p in properties)
+                    {
+                        SetProperty(p.Key, p.Value);
+                    }
+                }
+                Debug.Assert(entryPoint != null);
+                // Always check the static plugin factory table first, it takes precedence over the entryPoint specified
+                // in the plugin property value.
+                if (!_pluginFactories.TryGetValue(name, out IPluginFactory? pluginFactory))
```

Comment:
We no longer need to check `_pluginFactories` see previous comment

Reply:
Fixed.